projects/storefrontlib/src/cms-components/navigation/search-box/search-box.component.ts

```diff
@@ -107,17 +109,29 @@ export class SearchBoxComponent {
   /**
    * Closes the typehead searchbox.
    */
-  close(event: UIEvent): void {
-    if (!this.ignoreCloseEvent) {
-      this.searchBoxComponentService.toggleBodyClass(
-        'searchbox-is-active',
-        false
-      );
-      if (event && event.target) {
-        (<HTMLElement>event.target).blur();
+  close(event: UIEvent, force?: boolean): void {
+    // Use timeout to detect changes
+    setTimeout(() => {
+      if ((!this.ignoreCloseEvent && !this.isSearchboxFocused()) || force) {
+        this.searchBoxComponentService.toggleBodyClass(
+          'searchbox-is-active',
+          false
+        );
+        if (event && event.target) {
+          (<HTMLElement>event.target).blur();
+        }
       }
-    }
-    this.ignoreCloseEvent = false;
+      this.ignoreCloseEvent = false;
+    }, 0);
+  }
+
+  // Check if focus is on searchbox or result list elements
+  private isSearchboxFocused(): boolean {
+    return this.getResultElements().indexOf(this.getFocusedElement()) === -1 &&
+      this.winRef.document.querySelector('input[aria-label="search"]') !==
```

Comment:
this.winRef.document.querySelector('input[aria-label="search"]') === this.getFocusedElement()